### DISCUSSION TITLE
How does Textual render?

### DISCUSSION THREAD

**Discussion by ThatXliner** [Textualize/textual#514] [Q&A] [answered]
I was reading https://github.com/Textualize/textual/discussions/509 when I saw
> #### Reloading DockView content
> What is the best practice to reload the content of a dock view? Should I simply regenerate the whole DockView?

Does Textual optimize rendering like [React's virtual DOM](https://reactjs.org/docs/faq-internals.html#what-is-the-virtual-dom) (or even better, something like [Svelte](https://svelte.dev/blog/virtual-dom-is-pure-overhead))? Or does it not optimize at all? How does it render stuff and know what to change, etc, etc, etc??

**Comment by willmcgugan** [comment, marked as answer]
Well I don't know how these libraries do it, but Textual will compare one frame to the next to calculate deltas, and will batch changes over 50ms to reduce updates.

**Reply by ThatXliner** [reply]
So, if I’m not mistaken, it’s basically a VDOM.
